test/internal/internal.go

```diff
@@ -0,0 +1,223 @@
+	// test. See more at TODO LINK TO GINGKO LABEL EXAMPLES
+	labels []string
+
+	// Decorators are ginkgo decorators that can be applied to an individual test case
+	// Supported decorators are...
+	//
+	// Focus
+	//   The Focus decorator will force Ginkgo to run this test case and other test cases with the
+	//   Focus decorater while skipping all other test cases. Ginkgo does not considered any test
+	//   suite with a programatic focus decorator as passing the entirity of the test. While
+	//   the test suite might pass, the final exit status will be in error.
+	//
+	// Pending
+	//   The Pending decorator will instruct Ginkgo to skip the case. This is useful if a test is
+	//   under development, or perhaps is flaky.
+	//
+	// For more details, see
+	decorators []interface{}
+
+	directives []string
+
+	options TOptions
+}
+
+// TOptions let you configure things prior to a test running
+type TOptions struct {
+	// Create a storage profile for the test case
+	storageProfile *TStorageProfile
+}
+
+type TStorageProfile struct {
+	name string
+}
+
+func MakeTest(name string, directives ...string) *T {
+
+	// Extract a common set of labels from the directives
+	labels := make([]string, 0)
+	for _, directive := range directives {
+		args, err := dwdparse.BuildArgsMap(directive)
+		if err != nil {
+			panic(fmt.Sprintf("Test '%s' failed to parse provided directive '%s'", name, directive))
+		}
+
+		labels = append(labels, args["command"])
+
+		if len(args["type"]) != 0 {
+			labels = append(labels, args["type"])
+		}
+	}
+
+	return &T{
+		name:       name,
+		directives: directives,
+		labels:     labels,
+		decorators: make([]interface{}, 0),
+	}
+}
+
+func (t *T) WithStorageProfile(name string) *T {
+	t.options.storageProfile = &TStorageProfile{
+		name: name,
+	}
+
+	return t.WithLabels("storage-profile")
+}
+
+// To apply a set of labels for a particular test, use the withLables() method. Labels
+const (
+	Simple = "simple"
+)
+
+func (t *T) WithLabels(labels ...string) *T { t.labels = append(t.labels, labels...); return t }
+
+func (t *T) Focused() *T { t.decorators = append(t.decorators, Focus); return t }
+func (t *T) Pending() *T { t.decorators = append(t.decorators, Pending); return t }
+
+func (t *T) Name() string { return t.name }
+
+func (t *T) Args() []interface{} {
+	args := make([]interface{}, 0)
+
+	if len(t.labels) != 0 {
+		args = append(args, Labels(t.labels))
+	}
+
+	if len(t.decorators) != 0 {
+		args = append(args, t.decorators...)
+	}
+
+	return args
+}
+
+func (t *T) Labels() interface{} {
+	if len(t.labels) == 0 {
+		return nil
+	}
+
+	return Labels(t.labels)
+}
+
+func (t *T) Decorators() []interface{} {
+	if len(t.decorators) == 0 {
+		return nil
+	}
+
+	return t.decorators
+}
+
+func (t *T) Prepare(ctx context.Context, k8sClient client.Client) error {
+	o := t.options
+
+	if o.storageProfile != nil {
+		// Clone the placeholder profile
+		placeholder := &nnfv1alpha1.NnfStorageProfile{
```

Comment:
Do you have plans to allow for customization of profile's contents (e.g. if you wanted to tweak a setting to test)? 

Perhaps we could allow for the user to specify which profile to clone. You could pass in customized profiles for specific test cases. That would then require a directory of custom storage/container profiles.

Reply:
This is primarily focused on testing workflows from beginning to end and in my remaining time, I mostly wanted to conquer high-level code coverage. I didn't want to test the infinite options that storage profiles provide.

You can do any of which you propose by providing additional arguments to `WithStorageProfile()` or adding an entirely new TOption type